File: src/main/java/org/algohub/engine/type/ObjectMapperInstance.java
package org.algohub.engine.type;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;

public final class ObjectMapperInstance {
    public static final ObjectMapper INSTANCE = new ObjectMapper();

    private ObjectMapperInstance() {
        // static class
    }

    static {
        INSTANCE.registerModule(new Jdk8Module());
        INSTANCE.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
    }
}

File: src/main/kotlin/com/jalgoarena/ObjectMapperInstance.kt
package com.jalgoarena

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.module.SimpleModule
import com.fasterxml.jackson.module.kotlin.registerKotlinModule
import com.jalgoarena.type.ListNode

object ObjectMapperInstance {
    val INSTANCE: ObjectMapper = ObjectMapper()

    init {
        INSTANCE.registerKotlinModule()
        INSTANCE.setSerializationInclusion(JsonInclude.Include.NON_EMPTY)

        val customModule = SimpleModule()
        customModule.addDeserializer(ListNode::class.java, ListNode.Deserializer())
        INSTANCE.registerModule(customModule)
    }
}
